src/ProjectTemplates/Web.ProjectTemplates/content/BlazorWeb-CSharp/BlazorWeb-CSharp/Identity/PersistingRevalidatingAuthenticationStateProvider.cs

```diff
@@ -0,0 +1,106 @@
+using System.Diagnostics;
+using System.Security.Claims;
+using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Components.Authorization;
+using Microsoft.AspNetCore.Components.Server;
+using Microsoft.AspNetCore.Components.Web;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.Options;
+using BlazorWeb_CSharp.Client;
+using BlazorWeb_CSharp.Data;
+
+namespace BlazorWeb_CSharp.Identity;
+
+public class PersistingRevalidatingAuthenticationStateProvider : RevalidatingServerAuthenticationStateProvider
```

Comment:
Would be good to add a comment block above this class to indicate the basic ideas about what it does.